packages/firebase_ui_firestore/lib/src/query_builder.dart

```diff
@@ -500,6 +545,126 @@ class FirestoreListView<Document> extends FirestoreQueryBuilder<Document> {
             );
           },
         );
+
+  /// Shows a separator between list items just as in [ListView.separated]
+  FirestoreListView.separated({
+    Key? key,
+    required Query<Document> query,
+    required FirestoreItemBuilder<Document> itemBuilder,
+    int? pageSize,
+    FirestoreLoadingBuilder? loadingBuilder,
+    FirestoreLoadingIndicatorBuilder? loadingIndicatorBuilder,
+    FirestoreErrorBuilder? errorBuilder,
+    FirestoreEmptyBuilder? emptyBuilder,
+    required IndexedWidgetBuilder separatorBuilder,
+    Axis scrollDirection = Axis.vertical,
+    bool showLoadingIndicator = false,
+    bool reverse = false,
+    ScrollController? controller,
+    bool? primary,
+    ScrollPhysics? physics,
+    bool shrinkWrap = false,
+    EdgeInsetsGeometry? padding,
+    ChildIndexGetter? findChildIndexCallback,
+    bool addAutomaticKeepAlives = true,
+    bool addRepaintBoundaries = true,
+    bool addSemanticIndexes = true,
+    double? cacheExtent,
+    DragStartBehavior dragStartBehavior = DragStartBehavior.start,
+    ScrollViewKeyboardDismissBehavior keyboardDismissBehavior =
+        ScrollViewKeyboardDismissBehavior.manual,
+    String? restorationId,
+    Clip clipBehavior = Clip.hardEdge,
+  }) : super(
+          key: key,
+          query: query,
+          pageSize: pageSize ?? (showLoadingIndicator ? 20 : 10),
+          builder: (context, snapshot, _) {
+            if (snapshot.isFetching) {
+              return loadingBuilder?.call(context) ??
+                  const Center(child: CircularProgressIndicator());
+            }
+
+            if (snapshot.hasError && errorBuilder != null) {
+              return errorBuilder(
+                context,
+                snapshot.error!,
+                snapshot.stackTrace!,
+              );
+            }
+
+            if (snapshot.docs.isEmpty && emptyBuilder != null) {
+              return emptyBuilder(context);
+            }
+
+            final itemCount = snapshot.docs.length;
+
+            return ListView.separated(
+              itemCount: itemCount,
+              itemBuilder: (context, index) {
+                final isLastItem = index + 1 == itemCount;
+                if (!showLoadingIndicator && isLastItem && snapshot.hasMore) {
+                  snapshot.fetchMore();
+                }
+
+                final doc = snapshot.docs[index];
+                return showLoadingIndicator
+                    ? BuildAwareWidget(
+                        onBuild: () async {
+                          /// if we reached the end of the currently obtained
+                          /// items, we try to obtain more items
+                          if (isLastItem && snapshot.hasMore) {
```

Comment:
no need for comment, the code is pretty self-explanatory

Reply:
They were there in the original source code.
I'll delete these anyway.